pkg/cmd/cli/cmd/debug.go

```diff
@@ -264,6 +276,9 @@ func (o DebugOptions) Validate() error {
 	if len(names) == 0 {
 		return fmt.Errorf("the provided pod must have at least one container")
 	}
+	if (o.AsRoot || o.AsNonRoot) && o.AsUser > 0 {
+		return fmt.Errorf("you may not specify --as-root and --as-user=%d at the same time", o.AsUser)
```

Comment:
Shouldn't I be allowed to do `--as-root=false --as-user=12`?  I understand the distinction between the options, but I doubt that a user would expect the interaction to be this strict.